File: src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/MemoryUsage.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class MemoryUsage(
  val rss: Long,
  val heapTotal: Long,
  val heapUsed: Long,
  val external: Long,
  val arrayBuffers: Long,
)

File: agent/bindings/csharp/MemoryUsage.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class MemoryUsage
  {
    [JsonProperty(PropertyName = "rss")]
    public int Rss { get; set; }
    [JsonProperty(PropertyName = "heapTotal")]
    public int HeapTotal { get; set; }
    [JsonProperty(PropertyName = "heapUsed")]
    public int HeapUsed { get; set; }
    [JsonProperty(PropertyName = "external")]
    public int External { get; set; }
    [JsonProperty(PropertyName = "arrayBuffers")]
    public int ArrayBuffers { get; set; }
  }
}
